samples/cpp/3calibration.cpp

```diff
@@ -177,65 +178,49 @@ int main( int argc, char** argv )
-    const char* outputFilename = "out_camera_data.yml";
-    const char* inputFilename = 0;
+    float squareSize, aspectRatio;
+    string outputFilename;
+    string inputFilename = "";
 
     vector<vector<Point2f> > imgpt[3];
     vector<string> imageList;
 
+    cv::CommandLineParser parser(argc, argv,
+        "{w||}{h||}{s|1|}{o|out_camera_data.yml|}"
+        "{zt||}{a|1|}{p||}{@input||}");
+
     if(argc < 2)
     {
         help();
         return 1;
     }
-
-
-    for( i = 1; i < argc; i++ )
+    boardSize.width = parser.get<int>("w");
+    if (boardSize.width <= 0)
+        return fprintf( stderr, "Invalid board width\n" ), -1;
+    boardSize.height = parser.get<int>("h");
+    if (boardSize.height <= 0)
+        return fprintf( stderr, "Invalid board height\n" ), -1;
+    squareSize = parser.get<float>("s");
+    if (squareSize <= 0)
+        return fprintf( stderr, "Invalid board square width\n" ), -1;
+    aspectRatio = parser.get<float>("a");
+    if (parser.has("a"))
+        flags |= CALIB_FIX_ASPECT_RATIO;
+    if (aspectRatio <= 0)
+        return printf("Invalid aspect ratio\n" ), -1;
+    if (parser.has("zt"))
+        flags |= CALIB_ZERO_TANGENT_DIST;
+    if (parser.has("p"))
+        flags |= CALIB_FIX_PRINCIPAL_POINT;
+    if (!parser.check())
     {
-        const char* s = argv[i];
-        if( strcmp( s, "-w" ) == 0 )
-        {
-            if( sscanf( argv[++i], "%u", &boardSize.width ) != 1 || boardSize.width <= 0 )
-                return fprintf( stderr, "Invalid board width\n" ), -1;
-        }
-        else if( strcmp( s, "-h" ) == 0 )
-        {
-            if( sscanf( argv[++i], "%u", &boardSize.height ) != 1 || boardSize.height <= 0 )
-                return fprintf( stderr, "Invalid board height\n" ), -1;
-        }
-        else if( strcmp( s, "-s" ) == 0 )
-        {
-            if( sscanf( argv[++i], "%f", &squareSize ) != 1 || squareSize <= 0 )
-                return fprintf( stderr, "Invalid board square width\n" ), -1;
-        }
-        else if( strcmp( s, "-a" ) == 0 )
-        {
-            if( sscanf( argv[++i], "%f", &aspectRatio ) != 1 || aspectRatio <= 0 )
-                return printf("Invalid aspect ratio\n" ), -1;
-            flags |= CALIB_FIX_ASPECT_RATIO;
-        }
-        else if( strcmp( s, "-zt" ) == 0 )
-        {
-            flags |= CALIB_ZERO_TANGENT_DIST;
-        }
-        else if( strcmp( s, "-p" ) == 0 )
-        {
-            flags |= CALIB_FIX_PRINCIPAL_POINT;
-        }
-        else if( strcmp( s, "-o" ) == 0 )
-        {
-            outputFilename = argv[++i];
-        }
-        else if( s[0] != '-' )
-        {
-            inputFilename = s;
-        }
-        else
-            return fprintf( stderr, "Unknown option %s", s ), -1;
+        parser.printErrors();
+        return -1;
     }
+    outputFilename = parser.get<string>("o");
+    inputFilename = parser.get<string>("@input");
 
-    if( !inputFilename ||
+    if( !parser.check() ||
```

Comment:
The check for `inputFilename` emptyness should be added in this and some other files.